### PR TITLE
treewide: use ethtool_puts instead of memcpy

### PR DESCRIPTION
**Pull request by neheb**
The former is a safer and more readable version.

ping @Ansuel

### CODE REVIEW

**Comment by namiltd** [comment]
I think it will be even better using ethtool_puts
e.g. ethtool_puts&data, ag71xx_statistics[i].name); instead of ethtool_sprintf(&data, "%s", ag71xx_statistics[i].name);

**Comment by namiltd** [comment]
More simple is using two arguments ethtool_sprintf
e.g.
```
ethtool_sprintf(&data, ag71xx_statistics[i].name); 
```
instead of
```
ethtool_sprintf(&data, "%s", ag71xx_statistics[i].name);
```

**Comment by neheb** [comment]
Sure. Not available yet.

**Comment by ynezz** [comment]
> More simple is using two arguments ethtool_sprintf e.g.
> 
> ```
> ethtool_sprintf(&data, ag71xx_statistics[i].name); 
> ```

FYI it might be simpler, but [not correct](https://lore.kernel.org/all/202310101528.9496539BE@keescook/)

> Sure. Not available yet.

Well, it looks like [ethtool_puts](https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net-next.git/commit/?id=2a48c635fd9a) seems simple enough for backport/maintenance, so I'm wondering why to bother with already obsoleted `ethtool_sprintf` ?

File: target/linux/ramips/files/drivers/net/ethernet/ralink/ethtool.c [hunk 1]
```diff
@@ -159,11 +159,13 @@ static void fe_get_ringparam(struct net_device *dev,
...
-		memcpy(data, *fe_gdma_str, sizeof(fe_gdma_str));
-		break;
-	}
+	int i;
+
+	if (stringset != ETH_SS_STATS)
+		return;
+
+	for (i = 0; i < ARRAY_SIZE(fe_gdma_str); i++)
+		ethtool_sprintf(&data, "%s", fe_gdma_str[i]);
```
Comment (ynezz) [inline target/linux/ramips/files/drivers/net/ethernet/ralink/ethtool.c:168]: I would recommend avoiding such unrelated refactorings, because then its not "use ethtool_sprintf instead of memcpy" anymore, thus making the review harder. BTW if you look at `fe_get_sset_count()` you're actually introducing inconsistency, so I would keep it as it is.

**Comment by neheb** [comment]
> > More simple is using two arguments ethtool_sprintf e.g.
> > ```
> > ethtool_sprintf(&data, ag71xx_statistics[i].name); 
> > ```
> 
> FYI it might be simpler, but [not correct](https://lore.kernel.org/all/202310101528.9496539BE@keescook/)
> 
> > Sure. Not available yet.
> 
> Well, it looks like [ethtool_puts](https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net-next.git/commit/?id=2a48c635fd9a) seems simple enough for backport/maintenance, so I'm wondering why to bother with already obsoleted `ethtool_sprintf` ?

OTOH, Backporting a simple barely used function seens excessive.

File: target/linux/ramips/files/drivers/net/ethernet/ralink/ethtool.c (hunk 1 above)
Comment (neheb) [reply target/linux/ramips/files/drivers/net/ethernet/ralink/ethtool.c:168]: Upstream uses a combination of both styles. Probably better to reduce indentation.

File: target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/dsa.c [hunk 2]
```diff
@@ -941,8 +941,7 @@ static void rtl83xx_get_strings(struct dsa_switch *ds,
 		return;
 
 	for (int i = 0; i < ARRAY_SIZE(rtl83xx_mib); i++)
-		strncpy(data + i * ETH_GSTRING_LEN, rtl83xx_mib[i].name,
-			ETH_GSTRING_LEN);
+		ethtool_sprintf(&data, "%s", rtl83xx_mib[i].name);
```
Comment (raenye) [inline target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/dsa.c:944]:
```suggestion
		ethtool_sprintf(&data[i * ETH_GSTRING_LEN], "%s", rtl83xx_mib[i].name);
```

Comment (raenye) [reply target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/dsa.c:944]: Also, there's no `files-6.1` counterpart to this?

Comment (neheb) [reply target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/dsa.c:944]: Realtek has no 6.1 yet. This change seems wrong.

Comment (neheb) [reply target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/dsa.c:944]: Actually it’s completely wrong now that I look at it.

Comment (raenye) [reply target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/dsa.c:944]: Why even use `ethtool_sprintf` if you're not planning to update the source ptr?

Comment (neheb) [reply target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/dsa.c:944]: Because ethtool_sprintf already does so?

Comment (raenye) [reply target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/dsa.c:944]: But are you sure the `strlen` of each name is `ETH_GSTRING_LEN`?

Comment (neheb) [reply target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/dsa.c:944]: I don't think it matters. ethtool_sprintf uses ETH_GSTRING_LEN internally.

Also, see **https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/commit/?h=next-20231213id=4ddc1f1f7339e0a03bbf235a6670d497bbb22fb3**

https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/net/ethtool/ioctl.c?h=v6.6.7#n1990

**Comment by neheb** [comment]
@ynezz I did as requested and backported ethtool_puts instead. I put it as a backport of 6.10 (not out yet).
